app/admin/case_document.rb

```diff
@@ -1,9 +1,48 @@
-ActiveAdmin.register Case do
+ActiveAdmin.register CaseDocument do
   menu parent: "Cases", priority: 1
 
+  form title: "Document upload" do |f|
+    f.inputs "document" do
+      f.input :case
+      f.input :title
+      f.input :document, as: :file
+      f.input :document_type, as: :select, collection: %w[png jpg tiff], label: 'Convert to:'
+    end
+    f.actions
+  end
+
   controller do
     def permitted_params
       params.permit!
     end
+
+    def create
+      @case_document = CaseDocument.new(permitted_params[:case_document])
+      @case_document.document.attach(permitted_params[:case_document][:document])
+
+      @case_document.save!
+      # Assuming an image is uploaded, not any other file type
+      TranscodeImageJob.perform_later(@case_document, @case_document.document_type)
```

Comment:
You can also pass just case_document in the argument list instead of 2 arguments.

and then inside TranscodeImageJob

```
def perform(case_document)
  img = MiniMagick::Image.read(case_document.document.download)
  img = img.format(format_type)
  case_document.document.attach(
    io: File.open(img.path),
    filename: case_document.title,
    content_type: case_document.document_type,
  )
end
```

Reply:
Perfect, have amended this and will push now.